runtime/doc/autocmd.txt

```diff
@@ -470,11 +470,11 @@ CompleteDone			After Insert mode completion is done.  Either
 				    reason	Reason for completion being
 						done. Can be one of:
 						- "accept": completion was
-						  accepted using |complete_CTRL-Y|.
-						- "cancel": completion was cancelled
-						  using |complete_CTRL-E|, pressing
-						  a non-keyword character, or
-						  triggering a new completion.
+						  accepted by |complete_CTRL-Y|.
+						- "cancel": completion was
+						  stopped by |complete_CTRL-E.
+						- "abandon": completion was
```

Comment:
"discard" might fit better?

Reply:
Sure. My first reaction is that "discard" also somewhat implies explicit action, but that might looking to much into semantics.

Reply:
Actually for the "invoke completion() again" case, something like `restart` seems better. WDYT

Reply:
I'd understand "restart" as "start *this particular* completion again". This can be not the case when there is new `complete()` call.

Besides, I don't know of a (good) way to distinguish that particular case from others, so this is not really about this PR (which is for making "cancel" be strictly for explicit cancel via `<C-e>`). #28558 seems to have made an attempt, but it got its problems.